pangeo_forge_recipes/writers.py

```diff
@@ -79,12 +79,16 @@ def consolidate_metadata(store: MutableMapping) -> MutableMapping:
     import zarr
 
     if isinstance(store, fsspec.FSMap) and isinstance(store.fs, ReferenceFileSystem):
-        ref_path = store.fs.storage_args[0]
-        path = fsspec.get_mapper("reference://", fo=ref_path)
+        raise ValueError(
+            "Creating consolidated metadata for Kerchunk references has"
+            " no preformance benefits: Issue: "
+            " https://github.com/pangeo-forge/pangeo-forge-recipes/issues/675"
```

Comment:
Should we link to this issue here? It doesn't really explain or prove that there are no performance benefits.

I did a quick test (see this gist https://gist.github.com/abarciauskas-bgse/4cb32678f11aa04bcb281517a5e1862c) to see if I could tell if consolidating metadata mutates the store in any way and if there is performance benefits in that case. This was not an extensive test by any means but what I found was:

1. The json is the same between the two datasets except the consolidated version places all the zarr data in a `refs` key under the root, as opposed to directly under the root and the `refs` value holds all the data with an additional `.zmetadata` key which duplicates the data under the `.zarray` of the root object in the unconsolidated store.
2. Just as a sanity check and in case the libraries read the data differently when `consolidated=True` vs `consolidate=False`, I ran those performance tests a few times and, as noted in the notebook, the first time you generate the time series always takes the longest (4-5 seconds), regardless of whether the metadata is consolidated or not. Re-running the time series generation takes 1-1.5 seconds regardless as well.

Like I said, this was not a very exhaustive test but I don't think we have any reason to believe that performance would be different as the main point of consolidating metadata is just to read all the zarr metadata from one file which is already satisfied with the kerchunk reference. kerchunk is basically already consolidated metadata, but alongside the data itself (in a sense).

Reply:
Good call! I'll remove the link to the issue. Thanks for digging into this @abarciauskas-bgse.